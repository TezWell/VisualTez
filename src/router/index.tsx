import React from 'react';
import { Routes, Route, BrowserRouter } from 'react-router-dom';
import MarkdownLayout from 'src/components/MarkdownLayout';

import Page from 'src/components/Page';
import NotFound from 'src/pages/404/view';
import Editor from 'src/pages/editor';
import Landing from 'src/pages/landing/view';

import PrivacyMarkdown from 'src/pages/markdown/privacy.md';
import ContactMarkdown from 'src/pages/markdown/contact.md';
import { getBaseURL } from 'src/utils/path';

const Router = () => (
    <BrowserRouter>
        <Routes>
            <Route
                path={getBaseURL()}
                element={
                    <Page withNavigation withFooter>
                        <Landing />
                    </Page>
                }
            />
            <Route
                path={`${getBaseURL()}/editor`}
                element={
                    <Page withNavigation>
                        <Editor />
                    </Page>
                }
            />
            <Route
                path={`${getBaseURL()}/privacy`}
                element={
                    <Page withNavigation withFooter>
                        <MarkdownLayout markdown={PrivacyMarkdown} />
                    </Page>
                }
            />
            <Route
                path={`${getBaseURL()}/contact`}
                element={
                    <Page withNavigation withFooter>
                        <MarkdownLayout markdown={ContactMarkdown} />
                    </Page>
                }
            />
            <Route
                path="*"
                element={
                    <Page withNavigation withFooter>
                        <NotFound />
                    </Page>
                }
            />
        </Routes>
    </BrowserRouter>
);

export default Router;
