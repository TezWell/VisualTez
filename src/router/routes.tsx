import React from 'react';
import { Route, Routes } from 'react-router-dom';
import { Helmet } from 'react-helmet-async';

import Page from 'src/components/Page';
import EditorContext from 'src/context/Editor';
import AdaptableViewport from 'src/components/common/AdaptableViewport';

import PrivacyMarkdown from 'src/pages/markdown/privacy.md';
import ContactMarkdown from 'src/pages/markdown/contact.md';
import TermsMarkdown from 'src/pages/markdown/terms.md';
import MarkdownLayout from 'src/components/MarkdownLayout';
import { isDevelopment } from 'src/utils';

const EditorPage = React.lazy(() => import('src/pages/editor'));
const NodesPage = React.lazy(() => import('src/pages/nodes'));
const LandingPage = React.lazy(() => import('src/pages/landing/view'));
const NotFoundPage = React.lazy(() => import('src/pages/404'));
const DeployPage = React.lazy(() => import('src/pages/deploy'));

export const routes = [
    // Landing page
    {
        title: 'VisualTez',
        routeProps: {
            key: 'home',
            path: '/',
        },
        component: (
            <Page withNavigation withFooter>
                <LandingPage />
            </Page>
        ),
    },
    // Editor Page
    {
        disabled: !isDevelopment(),
        title: 'VisualTez - Editor',
        routeProps: {
            key: 'editor',
            path: '/editor',
        },
        component: (
            <AdaptableViewport minSize={768}>
                <Page withNavigation>
                    <EditorContext.Provider>
                        <EditorPage />
                    </EditorContext.Provider>
                </Page>
            </AdaptableViewport>
        ),
    },
    // Deploy Page
    {
        title: 'VisualTez - Deploy',
        routeProps: {
            key: 'deploy',
            path: '/deploy',
        },
        component: (
            <Page withNavigation>
                <DeployPage />
            </Page>
        ),
    },
    // Privacy Page
    {
        title: 'VisualTez - Privacy',
        routeProps: {
            key: 'privacy',
            path: '/privacy',
        },
        component: (
            <Page withNavigation withFooter>
                <MarkdownLayout markdown={PrivacyMarkdown} />
            </Page>
        ),
    },
    // Terms Page
    {
        title: 'VisualTez - Terms',
        routeProps: {
            key: 'terms',
            path: '/terms',
        },
        component: (
            <Page withNavigation withFooter>
                <MarkdownLayout markdown={TermsMarkdown} />
            </Page>
        ),
    },
    // Contact Page
    {
        title: 'VisualTez - Contact',
        routeProps: {
            key: 'contact',
            path: '/contact',
        },
        component: (
            <Page withNavigation withFooter>
                <MarkdownLayout markdown={ContactMarkdown} />
            </Page>
        ),
    },
    // Nodes Page
    {
        title: 'VisualTez - Nodes',
        routeProps: {
            key: 'nodes',
            path: '/nodes',
        },
        component: (
            <Page withNavigation withFooter>
                <NodesPage />
            </Page>
        ),
    },
    // Not Found Page
    {
        title: 'VisualTez - 404',
        routeProps: {
            key: '404',
            path: '*',
        },
        component: (
            <Page withNavigation withFooter>
                <NotFoundPage />
            </Page>
        ),
    },
];

const Routing = () => (
    <Routes>
        {routes
            .filter(({ disabled }) => !disabled)
            .map(({ title, routeProps, component }) => (
                <Route
                    {...routeProps}
                    element={
                        <>
                            <Helmet>
                                <title>{title}</title>
                            </Helmet>
                            {component}
                        </>
                    }
                />
            ))}
    </Routes>
);

export default Routing;
