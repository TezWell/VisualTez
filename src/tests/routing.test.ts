import { act, cleanup, waitFor } from '@testing-library/react';

import renderWithRouter from './renderWithRouter';
import { routes } from 'src/router/routes';

// Mock crypto.getRandomValues

Object.defineProperty(window, 'crypto', {
    value: { getRandomValues: () => new Uint32Array(10) },
});

afterEach(cleanup);

describe('Application Routing', () => {
    test.each(routes.map((route) => [route.title, route]))(
        `Test Route %s`,
        async (_, route: any) => {
            let path = route.routeProps.path;

            if (route.routeProps.path === '*') {
                path = '/not-found';
            }

            const { getByRole } = renderWithRouter({ path });
            const el = await waitFor(() => getByRole('main'), { timeout: 20000 });
            // Snapshot pages that do not use blockly
            if (!['/editor', '/deploy'].includes(route.routeProps.path)) {
                expect(el).toMatchSnapshot();
            }
            await waitFor(() => expect(document.title || 'FAILED').toMatch(route.title), {
                timeout: 20000,
            });
        },
        20000,
    );
});
