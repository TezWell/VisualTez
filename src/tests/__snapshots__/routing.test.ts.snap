// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application Routing Test Route VisualTez - 404 1`] = `
<main
  class="flex flex-col flex-1"
>
  <div
    class="flex-1 flex flex-col md:flex-row justify-center items-center"
  >
    <h1
      class="text-[4rem] md:text-[20rem] italic font-black tracking-widest diagonal-fractions"
    >
      404
    </h1>
    <p>
      Page 
      <b
        class="italic tracking-wider"
      >
        /
      </b>
       does not exist!
    </p>
  </div>
</main>
`;

exports[`Application Routing Test Route VisualTez - Contact 1`] = `
<main
  class="flex flex-col flex-1"
>
  <div
    class="prose dark:prose-invert md:prose-2xl container mx-auto pt-24 pb-24 pl-5 pr-5"
  >
    <p>
      contact.md
    </p>
  </div>
</main>
`;

exports[`Application Routing Test Route VisualTez - Nodes 1`] = `
<main
  class="flex flex-col flex-1"
>
  <div
    class="flex-1 inline-flex flex-row justify-center items-center flex-wrap container m-auto p-5"
  />
</main>
`;

exports[`Application Routing Test Route VisualTez - Privacy 1`] = `
<main
  class="flex flex-col flex-1"
>
  <div
    class="prose dark:prose-invert md:prose-2xl container mx-auto pt-24 pb-24 pl-5 pr-5"
  >
    <p>
      privacy.md
    </p>
  </div>
</main>
`;

exports[`Application Routing Test Route VisualTez - Terms 1`] = `
<main
  class="flex flex-col flex-1"
>
  <div
    class="prose dark:prose-invert md:prose-2xl container mx-auto pt-24 pb-24 pl-5 pr-5"
  >
    <p>
      terms.md
    </p>
  </div>
</main>
`;

exports[`Application Routing Test Route VisualTez 1`] = `
<main
  class="flex flex-col flex-1"
>
  <div
    class="pt-20 pb-20"
  >
    <div
      class="container px-3 mx-auto flex flex-wrap flex-col lg:flex-row items-center justify-evenly"
    >
      <div
        class="flex flex-col w-full md:w-4/5 lg:w-2/5 justify-center items-center lg:items-start text-center md:text-left"
      >
        <p
          class="uppercase tracking-loose w-full"
        >
          About the Project
        </p>
        <h1
          class="my-4 text-5xl font-bold leading-tight w-full"
        >
          VisualTez
        </h1>
        <p
          class="leading-normal text-2xl mb-8"
        >
          A visual programming tool for writing smart contracts in the Tezos blockchain.
        </p>
        <div
          class="inline-flex spacing-2"
        >
          <a
            href="/editor"
          >
            <button
              class="h-14 inline-flex items-center px-4 text-sm lg:text-lg bg-black dark:bg-white text-white dark:text-black font-bold rounded-full shadow-lg focus:outline-none focus:shadow-outline transform transition hover:scale-105 duration-300 ease-in-out"
            >
              Open Editor
            </button>
          </a>
          <a
            class="h-14 inline-flex items-center px-4 text-sm lg:text-lg bg-black dark:bg-white text-white dark:text-black font-bold rounded-full shadow-lg focus:outline-none focus:shadow-outline transform transition hover:scale-105 duration-300 ease-in-out ml-2"
            href="/docs"
            target="_blank"
          >
            Documentation
          </a>
        </div>
      </div>
      <div
        class="flex justify-center h-full w-full md:w-4/5 lg:w-2/5 py-6 text-center"
      >
        <img
          alt="example"
          class="h-full w-full"
          src="/assets/sample.svg"
        />
      </div>
    </div>
  </div>
</main>
`;
